File: retropreference/src/main/java/com/twiceyuan/library/typeHandler/LongHandler.java
package com.twiceyuan.library.typeHandler;

import android.content.SharedPreferences;

/**
 * Created by twiceYuan on 20/01/2017.
 *
 * Long handler
 */
public class LongHandler extends BaseTypeHandler<Long> {

    public LongHandler(SharedPreferences preferences) {
        super(preferences);
    }

    @Override
    public void setValue(String key, Long value) {
        mPreferences.edit().putLong(key, value).apply();
    }

    @Override
    public Long get(String key, Long defaultValue) {
        return mPreferences.getLong(key, defaultValue);
    }

    @Override
    public Long defaultValue() {
        return 0L;
    }
}

File: retropreference/src/main/java/com/twiceyuan/retropreference/typeHandler/LongHandler.kt
package com.twiceyuan.retropreference.typeHandler

import android.content.SharedPreferences

/**
 * Created by twiceYuan on 20/01/2017.

 * Long handler
 */
internal class LongHandler(preferences: SharedPreferences) : BaseTypeHandler<Long>(preferences) {

    override fun setValue(key: String, value: Long?) {
        mPreferences.edit().putLong(key, value ?: 0L).apply()
    }

    override operator fun get(key: String, defaultValue: Long?): Long? {
        return mPreferences.getLong(key, defaultValue ?: 0L)
    }

    override fun defaultValue(): Long? {
        return 0L
    }
}
